test/Datadog.Trace.ClrProfiler.Managed.Tests/Reflection/MethodBuilderTests.cs

```diff
@@ -13,13 +13,73 @@ public class MethodBuilderTests
         private readonly Assembly _thisAssembly = Assembly.GetExecutingAssembly();
         private readonly Type _testType = typeof(ObscenelyAnnoyingClass);
 
+        [Fact]
+        public void AmbiguousParameters_ClassASystemObject_CallsExpectedMethod()
+        {
+            var instance = new ObscenelyAnnoyingClass();
+            var p1 = new ClassA();
+            var p2 = new object();
+            var expected = MethodReference.Get(() => instance.Method(p1, p2));
+            var methodResult = Build<Action<object, object, object>>(expected.Name).WithParameters(p1, p2).Build();
+            methodResult(instance, p1, p2);
+            Assert.Equal(expected: expected.ToString(), instance.LastCall.MethodString);
+        }
+
+        [Fact]
+        public void AmbiguousParameters_ClassAClassA_CallsExpectedMethod()
+        {
+            var instance = new ObscenelyAnnoyingClass();
+            var p1 = new ClassA();
+            var p2 = new ClassA();
+            var expected = MethodReference.Get(() => instance.Method(p1, p2));
+            var methodResult = Build<Action<object, object, object>>(expected.Name).WithParameters(p1, p2).Build();
+            methodResult(instance, p1, p2);
+            Assert.Equal(expected: expected.ToString(), instance.LastCall.MethodString);
+        }
+
+        [Fact]
+        public void AmbiguousParameters_ClassAClassB_CallsExpectedMethod()
+        {
+            var instance = new ObscenelyAnnoyingClass();
+            var p1 = new ClassA();
+            var p2 = new ClassB();
+            var expected = MethodReference.Get(() => instance.Method(p1, p2));
+            var methodResult = Build<Action<object, object, object>>(expected.Name).WithParameters(p1, p2).Build();
+            methodResult(instance, p1, p2);
+            Assert.Equal(expected: expected.ToString(), instance.LastCall.MethodString);
+        }
+
+        [Fact]
+        public void AmbiguousParameters_ClassBClassC_CallsExpectedMethod()
+        {
+            var instance = new ObscenelyAnnoyingClass();
+            var p1 = new ClassB();
+            var p2 = new ClassC();
+            var expected = MethodReference.Get(() => instance.Method(p1, p2));
+            var methodResult = Build<Action<object, object, object>>(expected.Name).WithParameters(p1, p2).Build();
+            methodResult(instance, p1, p2);
+            Assert.Equal(expected: expected.ToString(), instance.LastCall.MethodString);
+        }
+
+        [Fact]
+        public void AmbiguousParameters_ClassCClassC_CallsExpectedMethod()
```

Comment:
This test uncovers the failure condition that I mentioned before, where the fallback has many matches, but no EXACT matches.